test/new-e2e/utils/e2e/client/agent.go

```diff
@@ -98,26 +102,33 @@ func (a *Agent) IsReady() (bool, error) {
 	return a.Status().isReady()
 }
 
+func (a *Agent) IsHealthy() (bool, error) {
+	output := a.Health()
+	return regexp.MatchString("Agent health: PASS", output)
+}
+
 // WaitForReady blocks up to one minute waiting for agent to be ready.
 // Retries every 100 ms up to one minute.
 // Returns error on failure.
 func (a *Agent) WaitForReady() error {
-	return a.WaitForReadyTimeout(1 * time.Minute)
+	return a.WaitForWithTimeout(a.IsReady, 1*time.Minute)
+}
+
+// WaitForHealthy blocks up to one minute waiting for agent to be healthy.
+// Retries every 100 ms up to one minute.
+// Returns error on failure.
+func (a *Agent) WaitForHealthy() error {
```

Comment:
Isn't `WaitForReady()` enough? Anyway I think we should have a single wait to make sure the Agent is ready.

Reply:
When testing locally the `WaitForReady()` does not indicate a healthy status, health is determined by registered component goroutines being [able to respond to ping timestamps](https://github.com/DataDog/datadog-agent/blob/main/pkg/status/health/health.go#L93-L108) via a channel it looks like which is separate from the status and takes longer (I think due to [initializing at time.Now() + ping frequency](https://github.com/DataDog/datadog-agent/blob/5f797db7bd6737429b0b01487eee80410889f2cc/pkg/status/health/health.go#L66))

I don't think I'm familiar enough with these commands to know which to use, do you have a preference? Status seems to be more comprehensive imo, we can add a wait of the 15s if this hits an error